### PR TITLE
Fixes the intent and messages ordering bug

### PR DESCRIPTION
**Pull request by goranambar**
Adds getNextOrderNumber for ordering fix and reads the complete intent object, with message templates

### CODE REVIEW

**Review by elliotmassen — APPROVED** [review]
This looks good! The only (very, very minor) thing I'm wondering is if the method should be `getHighestOrderNumber` and that incrementation becomes the responsibility of the caller instead? That way the method would be agnostic to its use, and could be used for other purposes too.

File: src/Conversation/DataClients/Relational/ConversationDataClient.php [hunk 1]
```diff
@@ -694,7 +694,8 @@ public function getIntentByUid(string $intentUid): Intent
         $conversation = $this->createConversationObject($conversationModel, $scenario);
         $scene = $this->createSceneObject($sceneModel, $conversation);
         $turn = $this->createTurnObject($turnModel, $scene);
-        $intent = $this->createIntentObject($intentModel);
+        $intent = $this->createIntentObjectTree($intentModel,
```
Comment (elliotmassen) [inline src/Conversation/DataClients/Relational/ConversationDataClient.php:697]: Is this related to this PR?

Comment (goranambar) [reply src/Conversation/DataClients/Relational/ConversationDataClient.php:697]: Kind of yes, because that function is used when binding value of Intent in the routes and if the whole tree isn't read, then the MessageTemplates aren't read and if they are not read then adding a new MessageTemplate will always have its ordering at 0.
